snuba/clickhouse/native.py

```diff
@@ -81,12 +82,26 @@ def __init__(
         self.client_settings = client_settings
 
         self.pool: queue.LifoQueue[Optional[Client]] = queue.LifoQueue(max_pool_size)
+        self.fallback_pool: queue.LifoQueue[Optional[Client]] = queue.LifoQueue(3)
         self.__gauge = ThreadSafeGauge(metrics, "connections")
```

Comment:
do we also want a gauge for fallback connections?

Reply:
ideally yes, but feel free to do that in a separate PR